src/components/ReportPage/index.jsx

```diff
@@ -65,6 +67,17 @@ class ReportPage extends PureComponent {
     return data;
   };
 
+  updateReportData = () => {
+    listenToSocketEvent(
+      'newAutomation',
+      (data) => {
+        const { reportData } = this.state;
+        const newData = [...data, ...reportData];
```

Comment:
Remove the spread from the `data` object, it breaks at some point and produces the following error:
`Uncaught TypeError: object is not iterable (cannot read property Symbol(Symbol.iterator)) at <anonymous>`

```suggestion
        const newData = [data, ...reportData];
```

Reply:
Thanks for the recommendation @09arnold, I think the error is due to the structure of the data been returned by the previous implementation on the backend. Like I explained in the DM earlier, we had to change the structure of the returned data to match the one already available on the frontend. The new implementation returns the data as an array and without the spread, we will be putting an array into `reportData` instead of an object.